### ISSUE TITLE
Why are Deflate Compression Levels ignored?

### ISSUE THREAD

**Issue by turbo** [mono/mono#6923]
This is related to overloads for classes which rely on DeflateStream and expose a CompressionLevel setting ([here's](https://msdn.microsoft.com/en-us/library/system.io.compression.compressionlevel(v=vs.110).aspx) a list for reference). Case in point: https://github.com/mono/mono/pull/1437.

I've been working on a project and kept wondering why `GZipStream` in particular would compress anything with the `CompressionLevel` set to None. A trip down to deflate revealed that it's just ignored:

```csharp
// snip
public DeflateStream(Stream stream, CompressionLevel compressionLevel, bool leaveOpen)
  : this(stream, compressionLevel, leaveOpen, false)
{
}

// snip

internal DeflateStream(Stream stream, CompressionLevel compressionLevel, bool leaveOpen, bool gzip)
  : this(stream, CompressionMode.Compress, leaveOpen, gzip)
{
  // ... what? 
}
```

Is there ... a technical reason for this? IMO the overload is *really* misleading. Compared to the MS API, this is a "breaking" change, is this documented somewhere?

**Comment by marek-safar** [comment]
Yeah, we are ignoring it at https://github.com/mono/mono/blob/f6364860459a488fdf7366a3f2573c14a45ddd49/mcs/class/System/System.IO.Compression/DeflateStream.cs#L101

**Comment by turbo** [comment]
Temporary workaround: Use the `GZipStream` provided by [haf/DotNetZip](https://github.com/haf/DotNetZip.Semverd). Maybe the implementation from that repo can be merged into Mono. The Zlib shared classes originate at Microsoft and still carry the MS-PL license. See 

https://github.com/haf/DotNetZip.Semverd/blob/738cd3835fcbd28310c738035c4b1a4f60dc0277/src/Zlib.Shared/GZipStream.cs#L1-L14

**Comment by marek-safar** [comment]
This should be addressed by #6974
